### PR TITLE
Pin email-validator dependency as specified in requirements.txt

### LINKED ISSUE #4009: [BUG] Issue with email-validator in test environment
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Since [email-validator](https://github.com/JoshData/python-email-validator) dependency has been bumped in 1.2.0, it refuses domain name like `@example.com`

I believe many projects (including ours) use such domain names in testing environments.
For this case, email-validator has given a parameter to be used in testing environments that disable the raise of an error if testing domain names are used in testing environments (`test_environment=True`)

The `requirements.txt` file of Pydantic specifies versions that are prior to 1.2.0 and thus working perfectly.
However, the `setup.py` file of Pydantic specifies version >=1.0.3 and the breaking change of email-validator has not been fixed

# Suggestion

I suggest that `setup.py` points the same version of `email-validator` as the one of `requirements.txt`

### PR DESCRIPTION
**Pull request by cgaunet**
## Change Summary

Pin email-validator dependency as specified in requirements.txt

## Related issue number

Fixes #4009 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by cgaunet** [comment]
Hello @samuelcolvin, a big thanks for your library 👏

This pull request aims at avoiding a breaking change in email-validator library, do not hesitate to ask for changes, if you could please review it, I'd be glad to help you :)

File: setup.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def extra(self):
         'typing-extensions>=3.7.4.3'
     ],
     extras_require={
-        'email': ['email-validator>=1.0.3'],
+        'email': ['email-validator==1.1.3'],
```
Comment (samuelcolvin) [inline setup.py:135]:
```suggestion
        'email': ['email-validator>=1.0.3,<1.2.0'],
```

is technically more correct.

Comment (cgaunet) [reply setup.py:135]: I strongly agree, I missed the syntax, thanks a lot 🙏

Comment (samuelcolvin) [reply setup.py]: just check the exact syntax, I might be wrong, but it's something like that.

**Comment by m3brown** [comment]
I added a comment to the issue related to this PR. While I don't believe merging this PR is a bad change (there _was_ a regression, hence why I found this PR!), I do suggest taking a moment to consider whether locking the max version is the appropriate response.

The new release (1.2.0) rejects more email addresses than it used to (including example.com), but I'm not sure that's a bad thing. Additionally, 1.2.0 provides some unicode/UTF-8 bugfixes that may be valuable to someone.

**Comment by samuelcolvin** [comment]
It seems 1.2.1 mostly reversed the changed and removed `example.com/org/net` from invalid domains.

With that, I don't think any change is necessary.

But thanks for altering me to this and proposing the change.

**Pull request closed without merging**